Common subdirectories: doop-r160113-bin/externals and doop-r160113-bin/externals
Common subdirectories: doop-r160113-bin/lib and doop-r160113-bin/lib
Common subdirectories: doop-r160113-bin/logic and doop-r160113-bin/logic
diff -u doop-r160113-bin/run doop-r160113-bin/run
--- doop-r160113-bin/run	2013-01-15 12:38:09.000000000 -0800
+++ doop-r160113-bin/run	2015-11-05 14:24:54.000000000 -0800
@@ -394,7 +394,7 @@
 fi
 
 # process the analysis argument
-if test -e logic/$1/analysis.logic; then
+if test -e ${DOOP_HOME}/logic/$1/analysis.logic; then
 	analysis=$1
 else
 	printf "${C_RED}ERROR${C_WHITE}: unsupported analysis \"%s\"${C_RESET}\n" "$1"
@@ -507,12 +507,12 @@
 {
 	mkdir -p tmp
 
-	preprocess logic/$analysis/declarations.logic tmp/$analysis-declarations.logic
-	preprocess logic/$analysis/delta.logic tmp/$analysis-delta.logic
-	preprocess logic/library/reflection-delta.logic tmp/reflection-delta.logic
-	preprocess logic/client/exception-flow-delta.logic tmp/exception-flow-delta.logic
-	preprocess logic/$analysis/analysis.logic tmp/$analysis.logic
-	preprocess logic/client/auxiliary-heap-allocations-delta.logic tmp/auxiliary-heap-allocations-delta.logic
+	preprocess $DOOP_HOME/logic/$analysis/declarations.logic tmp/$analysis-declarations.logic
+	preprocess $DOOP_HOME/logic/$analysis/delta.logic tmp/$analysis-delta.logic
+	preprocess $DOOP_HOME/logic/library/reflection-delta.logic tmp/reflection-delta.logic
+	preprocess $DOOP_HOME/logic/client/exception-flow-delta.logic tmp/exception-flow-delta.logic
+	preprocess $DOOP_HOME/logic/$analysis/analysis.logic tmp/$analysis.logic
+	preprocess $DOOP_HOME/logic/client/auxiliary-heap-allocations-delta.logic tmp/auxiliary-heap-allocations-delta.logic
 
 	# # Skolem Graph Creation
 	# if test $skolemGraph = "true"; then
@@ -600,7 +600,7 @@
 		timing $bloxbatch -db $cachedatabase -create -overwrite -blocks base
 
 		echo -n "loading fact declarations "
-		timing $bloxbatch -db $cachedatabase -addBlock -file logic/library/fact-declarations.logic
+		timing $bloxbatch -db $cachedatabase -addBlock -file $DOOP_HOME/logic/library/fact-declarations.logic
 
 		echo -n "loading facts "
 		rm -rf $(dirname $cachedatabase)/facts
@@ -637,7 +637,7 @@
 
 	if test "$sanity" = "true"; then
 		echo -n "loading sanity rules "
-		timing $bloxbatch -db $database -addBlock -file logic/library/sanity.logic
+		timing $bloxbatch -db $database -addBlock -file $DOOP_HOME/logic/library/sanity.logic
 	fi
 
 	echo -n "loading $analysis delta rules "
@@ -651,13 +651,13 @@
 
 	if test $transformInput = "true"; then
 	echo -n "preprocessing/transforming program facts: analysis"
-		timing $bloxbatch -db $database -addBlock -file logic/transform.logic
+		timing $bloxbatch -db $database -addBlock -file $DOOP_HOME/logic/transform.logic
 		echo -n "preprocessing/transforming program facts: transformation"
-		timing $bloxbatch -db $database -execute -file logic/transform-delta.logic
+		timing $bloxbatch -db $database -execute -file $DOOP_HOME/logic/transform-delta.logic
 		echo -n "preprocessing/transforming program facts: transformation"
-		timing $bloxbatch -db $database -execute -file logic/transform-delta.logic
+		timing $bloxbatch -db $database -execute -file $DOOP_HOME/logic/transform-delta.logic
 		echo -n "preprocessing/transforming program facts: transformation"
-		timing $bloxbatch -db $database -execute -file logic/transform-delta.logic
+		timing $bloxbatch -db $database -execute -file $DOOP_HOME/logic/transform-delta.logic
 	fi
 
 	# # Skolem Graph
@@ -746,7 +746,7 @@
 function reanalyze()
 {
 	echo "loading $analysis refinement-delta rules "
-	preprocess logic/$analysis/refinement-delta.logic tmp/$analysis-refinement-delta.logic
+	preprocess $DOOP_HOME/logic/$analysis/refinement-delta.logic tmp/$analysis-refinement-delta.logic
 
 	timing $bloxbatch -db $database -execute -file tmp/$analysis-refinement-delta.logic
 	timing $bloxbatch -db $database -exportCsv TempSiteToRefine -overwrite -exportDataDir tmp -exportFilePrefix $analysis-
@@ -789,16 +789,16 @@
 function run-stats()
 {
 	echo -n "loading statistics (simple) "
-	timing $bloxbatch -db $database -addBlock -file logic/library/statistics-simple.logic
+	timing $bloxbatch -db $database -addBlock -file $DOOP_HOME/logic/library/statistics-simple.logic
 	$bloxbatch -db $database -execute "+Stats:Runtime($elapsedTime, \"loading statistics (simple) time\")."
 
 	if test "$stats" = "true"; then
 		echo -n "loading statistics "
-		timing $bloxbatch -db $database -addBlock -file logic/library/statistics.logic
+		timing $bloxbatch -db $database -addBlock -file $DOOP_HOME/logic/library/statistics.logic
 		$bloxbatch -db $database -execute "+Stats:Runtime($elapsedTime, \"loading statistics time\")."
 		echo "sorting predicates ..."
 		start=`date +%s`
-		$bloxbatch -db $database -addBlock -file logic/library/statistics-sort.logic
+		$bloxbatch -db $database -addBlock -file $DOOP_HOME/logic/library/statistics-sort.logic
 
 		sortPredicate Stats:VarCount
 		sortPredicate Stats:InsensHeapVarCount
@@ -1011,7 +1011,7 @@
 link-result
 show-stats
 
-if test -e logic/$analysis/refinement-delta.logic; then
+if test -e $DOOP_HOME/logic/$analysis/refinement-delta.logic; then
 	reanalyze
 
 	run-stats
Common subdirectories: doop-r160113-bin/tests and doop-r160113-bin/tests
